File: lib/domain/field/results/farm_fields_list.rb
# frozen_string_literal: true

module Domain
  module Field
    module Results
      # 農場に紐づく圃場一覧ユースケースの戻り（表現形式に依存しない）。
      class FarmFieldsList
        attr_reader :farm, :fields, :turbo_stream_subscription

        def initialize(farm:, fields:, turbo_stream_subscription: nil)
          @farm = farm
          @fields = fields
          @turbo_stream_subscription = turbo_stream_subscription ||
            Domain::Shared::Dtos::TurboStreamSubscription.for_farm(farm.id)
        end
      end
    end
  end
end

File: crates/agrr-domain/src/field/results/farm_fields_list.rs
use crate::field::entities::FieldEntity;
use crate::field::results::FarmRecord;

/// Ruby: `Domain::Field::Results::FarmFieldsList`
#[derive(Debug, Clone, PartialEq)]
pub struct FarmFieldsList {
    pub farm: FarmRecord,
    pub fields: Vec<FieldEntity>,
}

impl FarmFieldsList {
    pub fn new(farm: FarmRecord, fields: Vec<FieldEntity>) -> Self {
        Self { farm, fields }
    }
}
